cli/tests/pcluster/validators/test_ec2_validators.py

```diff
@@ -457,3 +458,69 @@ def test_placement_group_validator(
     )
     actual_failures = PlacementGroupIdValidator().execute(placement_group=placement_group)
     assert_failure_messages(actual_failures, expected_message)
+
+
+@pytest.mark.parametrize(
+    "capacity_reservation_instance_type, capacity_reservation_availability_zone, "
+    "instance_type, subnet_availability_zone, expected_message",
+    [
+        ("c5.xlarge", "us-east-1a", "c5.xlarge", "us-east-1a", None),
+        # Wrong instance type
+        (
+            "m5.xlarge",
+            "us-east-1a",
+            "c5.xlarge",
+            "us-east-1a",
+            "Capacity reservation .* must has the same instance type as c5.xlarge.",
+        ),
+        # Wrong availability zone
+        (
+            "c5.xlarge",
+            "us-east-1b",
+            "c5.xlarge",
+            "us-east-1a",
+            "Capacity reservation .* must use the same availability zone as subnet",
+        ),
+        # Both instance type and availability zone are wrong
+        (
+            "m5.xlarge",
+            "us-east-1b",
+            "c5.xlarge",
+            "us-east-1a",
+            "Capacity reservation .* must has the same instance type as c5.xlarge.",
+        ),
+        (
+            "m5.xlarge",
+            "us-east-1b",
+            "c5.xlarge",
+            "us-east-1a",
+            "Capacity reservation .* must use the same availability zone as subnet",
+        ),
+    ],
+)
+def test_capacity_reservation_validator_validator(
```

Comment:
Can we also add the test for CapacityReservationResourceGroupValidator?

Reply:
Done